### PR TITLE
add cake build

### PR DESCRIPTION
**Pull request by sphiecoh**

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    .Does(() =>
+{
+    var settings = new DotNetCorePackSettings
+    {
+        Configuration = "Release",
+        OutputDirectory = buildArtifacts,
+        NoBuild = true,
+        VersionSuffix = versionSuffix
+    };
+    var projects = GetFiles("./src/**/*.csproj");
```
Comment (jchannon) [inline build.cake]: That should probably hardcode the path  and just call `DotNetCorePack("./src/Botwin", settings);`

File: build/version.props [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<Project>
+ <PropertyGroup>
+    <VersionPrefix>1.0.0</VersionPrefix>
+    <VersionSuffix>beta2</VersionSuffix>
+    <VersionSuffix Condition=" '$(VersionSuffix)' != '' AND '$(BuildNumber)' != '' ">$(VersionSuffix)-$(BuildNumber)</VersionSuffix>
```
Comment (jchannon) [inline build/version.props]: Where does BuildNumber come from?

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env bash
```
Comment (jchannon) [inline build.sh]: guessing this is a copy/paste?

File: build.ps1 [hunk 4]
```diff
@@ -0,0 +1,228 @@
+##########################################################################
```
Comment (jchannon) [inline build.ps1]: guessing this is a copy/paste?

File: build.sh [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    fi
+fi
+
+# Restore tools from NuGet.
+pushd "$TOOLS_DIR" >/dev/null
+if [ ! -f "$PACKAGES_CONFIG_MD5" ] || [ "$( cat "$PACKAGES_CONFIG_MD5" | sed 's/\r$//' )" != "$( $MD5_EXE "$PACKAGES_CONFIG" | awk '{ print $1 }' )" ]; then
+    find . -type d ! -name . | xargs rm -rf
+fi
+
+mono "$NUGET_EXE" install -ExcludeVersion
```
Comment (jchannon) [inline build.sh]: Do we need mono? Ideally not? @devlead

File: build.sh [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+# Make sure that Cake has been installed.
+if [ ! -f "$CAKE_EXE" ]; then
+    echo "Could not find Cake.exe at '$CAKE_EXE'."
+    exit 1
+fi
+
+# Start Cake
+if $SHOW_VERSION; then
+    exec mono "$CAKE_EXE" -version
```
Comment (jchannon) [inline build.sh]: Can cake run without mono? @devlead

File: build.sh (hunk 3 above)
Comment (Sphiecoh) [inline build.sh]: default cake bootstrapper

File: build.ps1 (hunk 4 above)
Comment (Sphiecoh) [inline build.ps1]: default cake bootstrapper

File: build.sh (hunk 5 above)
Comment (devlead) [inline build.sh]: Yes It's possible to run without mono, you'll have to fetch cake using curl&unzip something like
This example: https://github.com/devlead/BitbucketPipelinesShield/blob/master/build.sh

**Comment by Sphiecoh** [comment]
I have removed mono

**Comment by jchannon** [comment]
I had a play and have it working with codeship currently.  Will keep this
PR open for a bit just to see how we get on with codeship first. hope thats
ok?

**Pull request closed without merging**